Swift/Query.swift

```diff
@@ -124,6 +125,28 @@ public class Query {
         return listenerToken
     }
     
+    /// Returns a Combine publisher that emits `QueryChange` events whenever the query
+    /// result changes.
+    /// - Note: The emitted `QueryChange` contains the entire new result set of the
+    ///         query, **not just the delta**.
+    ///
+    /// - Parameter queue: The `DispatchQueue` for event delivery. Defaults to the main queue.
+    /// - Returns: A `PassthroughSubject<QueryChange, Never>` that emits query changes
+    /// - Note: Only available on  iOS 13.0 and later.
+    @available(iOS 13.0, *)
+    public func changePublisher(on queue: DispatchQueue = .main) -> AnyPublisher<QueryChange, Never> {
+        let subject = PassthroughSubject<QueryChange, Never>()
+        
+        let token = self.addChangeListener(withQueue: queue) { change in
+            subject.send(change)
+        }
+
+        return subject
+            .receive(on: queue)
+            .handleEvents(receiveCancel: { token.remove() })
+            .eraseToAnyPublisher()
```

Comment:
just out of curiosity, what does "eraseToAnyPublisher" do?

Reply:
it hides the publisher type converting it to AnyPublisher for subscription. It is future proof in case we change any implementation.